lark/tools/__init__.py

```diff
@@ -22,17 +23,17 @@
 lalr_argparser.add_argument('-v', '--verbose', action='count', default=0, help="Increase Logger output level, up to three times")
 lalr_argparser.add_argument('-s', '--start', action='append', default=[])
 lalr_argparser.add_argument('-l', '--lexer', default='contextual', choices=('basic', 'contextual'))
-k = {'encoding': 'utf-8'} if sys.version_info > (3, 4) else {}
-lalr_argparser.add_argument('-o', '--out', type=FileType('w', **k), default=sys.stdout, help='the output file (default=stdout)')
-lalr_argparser.add_argument('grammar_file', type=FileType('r', **k), help='A valid .lark file')
-
-for f in flags:
-    if isinstance(f, tuple):
-        options.append(f[1])
-        lalr_argparser.add_argument('-' + f[0], '--' + f[1], action='store_true')
-    else:
-        options.append(f)
-        lalr_argparser.add_argument('--' + f, action='store_true')
+encoding: Optional[Literal['utf-8']] = 'utf-8' if sys.version_info > (3, 4) else None
+lalr_argparser.add_argument('-o', '--out', type=FileType('w', encoding=encoding), default=sys.stdout, help='the output file (default=stdout)')
+lalr_argparser.add_argument('grammar_file', type=FileType('r', encoding=encoding), help='A valid .lark file')
+
+for flag in flags:
+    if isinstance(flag, tuple):
+        options.append(flag[1])
+        lalr_argparser.add_argument('-' + flag[0], '--' + flag[1], action='store_true')
+    elif isinstance(flag, str):
+        options.append(flag)
+        lalr_argparser.add_argument('--' + flag, action='store_true')
 
```

Comment:
Now flags that are neither, will simply be ignored. That's not great. Instead, we should throw an error.

Reply:
c34b3c576fd63549b3541b004232a9e561b4831b